Testscripts/Linux/verify_uio_driver.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+########################################################################
+#
+# Description:
+#       This script was created to verify the uio_hv_generic module.
+#
+#       The test performs the following steps:
+#         1. Assign a NIC to uio_hv_generic driver and check the uio device
+#         2. Restore to hv_netvsc and check the eth1
+#
+########################################################################
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+#######################################################################
+# Pre-settings & Functions
+#######################################################################
+# Check distro and kernel version
+case "$DISTRO" in
+    redhat_7)
+        if ! CheckVMFeatureSupportStatus "3.10.0-692"; then
+            LogMsg "INFO: this kernel version does not support uio feature, skip test"
+            UpdateSummary "INFO: this kernel version does not support uio feature, skip test"
+            SetTestStateSkipped
+            exit 0
+        fi
+        ;;
+    redhat_8)
+        UpdateSummary "$DISTRO support uio feature. Continue."
+        ;;
+    *)
+        LogMsg "$Distro is not supported in this case. Skip."
+        SetTestStateAborted
+        exit 0
+        ;;
+esac
+
+# Get device_id and class_id
+device_id=$(cat /sys/class/net/eth1/device/device_id|tr -d '{}')
+class_id=$(cat /sys/class/net/eth1/device/class_id|tr -d '{}')
+
+function register_uio() {
+    LogMsg "Register eth1 to uio"
+    output=$(modprobe uio_hv_generic 2>&1)
+    lsmod | grep uio_hv_generic
+    CheckPass "Verify uio_hv_generic is loaded"
+
+    [ -z "$output" ]
+    CheckPass "Verify no output while modprobing uio_hv_generic" "Output: $output"
+
+    echo ${class_id} > /sys/bus/vmbus/drivers/uio_hv_generic/new_id
+    echo ${device_id} > /sys/bus/vmbus/drivers/hv_netvsc/unbind
+    echo ${device_id} > /sys/bus/vmbus/drivers/uio_hv_generic/bind
+    # Verify /dev/uio0 exists
+    [ -e /dev/uio0 ]
+    CheckPass "Verify /dev/uio0 exists"
+}
+
+function recovery() {
+    LogMsg "Restore to hv_netvsc"
+    echo ${device_id} > /sys/bus/vmbus/drivers/uio_hv_generic/unbind
+    echo ${device_id} > /sys/bus/vmbus/drivers/hv_netvsc/bind
+    # Verify eth1 exists
+    [ ! -e /dev/uio0 ]
+    CheckPass "Verify /dev/uio0 is removed"
+    # Verify eth1 doesn't exist
+    ip addr show eth1
```

Comment:
exist or not exist?

Reply:
Sorry for the wrong messages. I'll update it. Thanks!